pyspecdata/core.py

```diff
@@ -2052,45 +2096,56 @@ def set_error(self, *args):
         .. todo::
                 several options below -- enumerate them in the documentation
         """
-        if (len(args) == 1) and np.isscalar(args[0]):
-            if args[0] == 0:
-                args = (np.zeros_like(self.data),)
+        if len(args) == 1:
+            if np.isscalar(args[0]):
+                logger.debug("converting scalar")
+                if args[0] == 0:
+                    args = (np.zeros_like(self.data),)
+                else:
+                    args = (np.ones_like(self.data) * args[0],)
+            logger.debug(
+                "type is now "
+                + str(type(args[0]))
+                + "is numpy?: "
+                + str(isinstance(args[0], np.ndarray))
+            )
+            if isinstance(args[0], np.ndarray):
+                self.data_error = np.reshape(args[0], np.shape(self.data))
+            elif isinstance(args[0], list):
+                self.data_error = np.reshape(
+                    np.array(args[0]), np.shape(self.data)
+                )
+            elif args[0] is None:
+                self.data_error = None
+            elif np.isscalar(args[0]):
+                self.data_error = args[0]
             else:
```

Comment:
this doesn't make sense didn't we already have an if the args[0] is scalar above where we convert it to array?